public/app/core/components/form_dropdown/form_dropdown.ts

```diff
@@ -214,7 +225,9 @@ export class FormDropdownCtrl {
 
     var typeahead = this.inputElement.data('typeahead');
     if (typeahead) {
-      this.inputElement.val('');
+      if (!this.allowCustom) {
+        this.inputElement.val('');
+      }
```

Comment:
this is a problem, if we do not remove the current value the typeahead wont show alternatives.